packages/client/src/createAttachmentReader.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  Attachment,
+  AttachmentMetadata,
+  AttachmentSignature,
+} from "@osdk/client.api";
+import { Ontologies } from "@osdk/internal.foundry";
+import type { MinimalClient } from "./MinimalClientContext.js";
+
+export function createAttachmentReader(
```

Comment:
At a future date we may want to make an internal-only class that implements this or a prototype so we don't produce a ton of churn on "hidden classes".